app/src/main/java/com/example/wnews/views/home/news/presenter/NewsPresenter.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.wnews.views.home.news.presenter
+
+import android.util.Log
+import android.widget.Adapter
+import androidx.lifecycle.MutableLiveData
+import androidx.recyclerview.widget.RecyclerView
+import com.example.wnews.models.ListNews
+import com.example.wnews.models.News
+import com.example.wnews.models.UserAuth
+import com.example.wnews.providers.RetrofitProvider
+import com.example.wnews.views.home.news.NewsView
+import com.example.wnews.views.home.news.adapter.NewsAdapter
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class NewsPresenter(val newsView: NewsView) {
+
+    var arrayListNews = arrayListOf<News>()
+        private set
+
+    var loading = false
+        private set
+
+    fun saveAsFavorite() {}
+
+    fun clearArrayList() {
+        arrayListNews = arrayListOf()
+    }
+
+    fun onResponseNews(page: Int, userAuth: UserAuth) {
+        loading = true
+        newsView.showProgressBar(loading)
+
+
+        val call: Call<ListNews> = RetrofitProvider.newsService.getNews(
+            page,
+            userAuth.token,
+            userAuth.client,
+            userAuth.uid
+        )
+        call.enqueue(object : Callback<ListNews?> {
+
+            override fun onResponse(
+                call: Call<ListNews?>?,
+                response: Response<ListNews?>?
+            ) {
+
+                if (response!!.isSuccessful) {
+
+                    val items = response.body()
+
+
+                    if (!items!!.page.isNullOrEmpty()) {
+                        items.page.forEach { item ->
+
+                            arrayListNews.add(
+                                News(
+                                    item.newsId,
+                                    item.title,
+                                    item.detail,
+                                    item.imageUrl
+                                )
+                            )
+                        }
+                    }
+
+                } else if (response.errorBody() != null) {
+
+                }
+
+                loading = false
+
+                newsView.changeData()
+
+                newsView.showProgressBar(loading)
+
+            }
+
+            override fun onFailure(call: Call<ListNews?>?, t: Throwable?) {
+                loading = false
+
```

Comment:
Maybe we could improve the error handling here. If there is some sort of issue with the backend or with the network connection the user would just see the spinner disappear. 

What do you think about adding some `TextView` to the middle of the layout that display something like "We couldn't retrieve the  news at this time, please try again later" and displaying that mesage only on this case?